rest_framework/serializers.py

```diff
@@ -801,9 +801,8 @@ def raise_errors_on_nested_writes(method_name, serializer, validated_data):
     #     ...
     #     address = serializer.CharField('profile.address')
     assert not any(
-        '.' in field.source and
-        (key in validated_data) and
-        isinstance(validated_data[key], (list, dict))
```

Comment:
Added this back, since it can catch cases where e.g. a serializer has a field with source 'profile' and another field with source 'profile.address'.